CSS/CSS.sublime-syntax

```diff
@@ -92,1903 +432,1490 @@ contexts:
+    - match: \}
+      scope: punctuation.section.block.end.css
+      pop: 1
+    - include: rule-list-body
 
   rule-list:
-    - match: '\{'
-      scope: punctuation.section.property-list.css
-      push:
-        - meta_scope: meta.property-list.css
-        - match: '(?=\s*\})'
-          pop: true
-        - include: rule-list-body
+    - match: \{
+      scope: punctuation.section.block.begin.css
+      push: rule-list-content
 
-  rule-list-body:
-    - include: comment-block
-    - match: (?=[-a-z])
-      push:
-        - meta_scope: meta.property-name.css
-        - match: $|(?![-a-z])
-          pop: true
-        - include: vendor-prefix
-        - include: deprecated-property
-        - include: custom-property-name
-        - include: counter-property
-        - include: font-property
-        - include: other-property
-    - match: (:)([ \t]*)
-      captures:
-        1: punctuation.separator.key-value.css
-        2: meta.property-value.css
-      push: property-value-content
+  rule-list-content:
+    - meta_scope: meta.property-list.css meta.block.css
+    - match: \}
+      scope: punctuation.section.block.end.css
+      pop: 2
+    - include: rule-list-body
 
-  property-value-content:
-    - meta_content_scope: meta.property-value.css
-    - include: property-value-end
+  rule-list-body:
+    # Note: This context is used by HTML.sublime-syntax
+    - include: comments
+    - include: property-identifiers
     - include: property-values
 
-  property-value-end:
-    - match: \s*(;)|(?=[})])
-      captures:
-        1: punctuation.terminator.rule.css
-      pop: true
+###[ PROPERTY IDENTIFIERS ]####################################################
+
+  property-identifiers:
+    - match: (?=[-a-z])
+      push: property-identifier-content
+
+  property-identifier-content:
+    - meta_scope: meta.property-name.css
+    - include: vendor-prefixes
+    - include: font-property
+    - include: counter-property
+    - include: other-property
+    - include: custom-property
+    - include: deprecated-property
+    - include: immediately-pop
 
   counter-property:
     # https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Lists_and_Counters/Using_CSS_counters
     - match: \bcounter-(increment|reset|set)(?!-)\b
       scope: support.type.property-name.css
-      push:
-        - match: (:)([ \t]*)
-          captures:
-            1: punctuation.separator.key-value.css
-            2: meta.property-value.css
-          set:
-            - property-value-content
-            - counter-identifier
-        - match: ''
-          pop: true
+      set: counter-property-value
 
-  counter-identifier:
-    - match: '{{ident}}'
-      scope: variable.other.counter.css
-      pop: true
-    - match: (?=\S)
-      pop: true
+  # Custom Properties
+  # https://drafts.csswg.org/css-variables/#typedef-custom-property
+  custom-properties:
+    - match: (--)({{nmchar}}+)
+      scope: support.type.custom-property.css
```

Comment:
I am not very happy about `support.type` at all, but also the fact those are stacked on each other here.

Thoughts?